Source/WebCore/platform/libwebrtc/LibWebRTCVPXVideoDecoder.cpp

```diff
@@ -169,6 +170,7 @@ LibWebRTCVPXInternalVideoDecoder::LibWebRTCVPXInternalVideoDecoder(LibWebRTCVPXV
     : m_outputCallback(WTFMove(outputCallback))
     , m_internalDecoder(createInternalDecoder(type))
     , m_useIOSurface(config.pixelBuffer == VideoDecoder::HardwareBuffer::Yes)
+    , m_isFullRange((type == LibWebRTCVPXVideoDecoder::Type::VP8 || type == LibWebRTCVPXVideoDecoder::Type::VP9 || type == LibWebRTCVPXVideoDecoder::Type::VP9_P2) && config.description.size() >= 12 && (*(config.description.data() + 6) & 0x1))
```

Comment:
This is rather cryptic, can we introduce a routine for checking full range and taking a span?
This will be useful when we move the VTB VP9 decoder in WebCore.
Also, I would use `config.description[6]` and maybe set it to a `bitDepthChromaAndRange` variable.

Reply:
Additionally, this doesn't appear to handle the case that the color information is parsed out of the container file.

Can we just add `PlatformVideoColorSpace colorSpace;` to the `VideoDecoder::Config` struct?

Reply:
> Additionally, this doesn't appear to handle the case that the color information is parsed out of the container file.
> 
> Can we just add `PlatformVideoColorSpace colorSpace;` to the `VideoDecoder::Config` struct?

I'll take this as a follow-up, as it's a big change and require to modify the SourceBufferParserWebM to provide this information.

Reply:
I opened https://bugs.webkit.org/show_bug.cgi?id=282741 ; though I will be adding construction of the colorspace in this PR .